railties/lib/rails/generators.rb

```diff
@@ -156,12 +156,25 @@ def self.invoke(namespace, args=ARGV, config={})
         args << "--help" if args.empty? && klass.arguments.any?(&:required?)
         klass.start(args, config)
       else
-        options     = sorted_groups.map(&:last).flatten
-        suggestions = options.sort_by {|suggested| levenshtein_distance(namespace.to_s, suggested) }.first(3)
-        msg =  "Could not find generator '#{namespace}'. "
-        msg << "Maybe you meant #{ suggestions.map {|s| "'#{s}'"}.join(" or ") }\n"
-        msg << "Run `rails generate --help` for more options."
-        puts msg
+        unless sorted_groups.group_by(&:size).min.first > namespace.length
+          options     = sorted_groups.map(&:last).flatten
+          suggestions = options.select do |option|
+            option.split(/[:_]/).any? { |option_segment|
+              namespace.to_s.split(/[:_]/).any? { |name_segment|
```

Comment:
The current implementation actually doesn't have the max number of suggestions (originally it's limited to 3), so although it will lists up whatever coffee-prefixed (`coffee:foo`, `coffee:bar` or `coffee:baz`), it's still along with anything ending with `assets` or whatever 1 char different from <kbd>assetss</kbd>. Perhaps we should look at the total distance of once-split inputs though.